src/api/privateApi.js

```diff
@@ -186,113 +186,56 @@ function serialiseHttpData(data) {
 }
 
 /**
- * @method checkRequest
+ * @method createRequestObject
+ * @param requestMethod
  * @param requestType
  * @param options
  * @private
  *
- * @return method string
+ * @return request Object
  */
 
-function checkRequest(requestType, options) {
-	return this.parseOfflineRequests(requestType, options).requestMethod;
-}
-
-function transformGETRequest(baseRequestObject, requestType, options) {
-	const requestMethod = 'GET';
-	const requestUrlBase = `${getFullUrl.call(this)}/api/${requestType}`;
-	const requestUrl = Object.keys(options).length
-		? requestUrlBase + serialiseHttpData.call(this, options, requestMethod)
-		: requestUrlBase;
-	const requestParams = options;
-	return Object.assign({}, baseRequestObject, {
-		requestMethod,
-		requestUrl,
-		requestParams,
-	});
-}
-
-function transformPUTOrPOSTRequest(baseRequestObject, requestType, options) {
-	const transformRequest = this
-		.parseOfflineRequests(requestType, options)
-		.checkOfflineRequestBefore();
-
-	return (transformRequest.requestUrl === 'transactions' || transformRequest.requestUrl === 'signatures')
-		? Object.assign({}, {
-			requestUrl: `${getFullUrl.call(this)}/peer/${transformRequest.requestUrl}`,
-			nethash: this.nethash,
-			requestMethod: 'POST',
-			requestParams: transformRequest.params,
-		})
-		: Object.assign({}, baseRequestObject, {
-			requestUrl: `${getFullUrl.call(this)}/api/${transformRequest.requestUrl}`,
-			requestMethod: transformRequest.requestMethod,
-			requestParams: options,
-		});
-}
-
-/**
- * @method changeRequest
- * @param requestType
- * @param options
- * @private
- *
- * @return httpRequest object
- */
+function createRequestObject(requestMethod, requestType, options) {
+	const requestUrl = requestMethod === 'GET'
+		? `${getFullUrl.call(this)}/api/${requestType}${serialiseHttpData.call(this, options, requestMethod)}`
+		: `${getFullUrl.call(this)}/api/${requestType}`;
 
-function changeRequest(requestType, options) {
-	const defaultRequestObject = {
-		requestMethod: '',
-		requestUrl: '',
-		nethash: '',
-		requestParams: '',
+	return {
+		method: requestMethod,
+		url: requestUrl,
+		headers: this.nethash,
```

Comment:
I understand the idea and would support it, where do you have in mind nethash should be passed in first time?